src/qml/Toast.qml

```diff
@@ -50,6 +51,31 @@ Popup {
     color: "#66212121"
     radius: 4
 
+    ProgressBar {
+      id: animationProgressBar
+      padding: 2
+      anchors.fill: parent
+      visible: animationEnabled
+      z: toastRow.z - 1
+      value: animationTimer.position / toastTimer.interval
+
+      background: Item {
+        anchors.fill: parent
+      }
+
+      contentItem: Item {
+        anchors.fill: parent
+
+        Rectangle {
+          width: animationProgressBar.visualPosition * parent.width
+          height: parent.height
+          radius: 2
+          color: "#8080cc28"
```

Comment:
Use the Theme.mainColor and the Qt.hsla property that allows to tweak alpha instead of a hard-coded value